### ISSUE TITLE
Clarification Needed: Unable to Locate `MemGPT` Class Definition in Repository

### ISSUE THREAD

**Issue by hjhmuser** [letta-ai/letta#1360]
Hello,

I am currently attempting to replicate and understand the long document QA experiments mentioned in your paper using the source code provided in this repository. However, I have encountered a stumbling block that I hope to get some clarification on.

While going through the source code, especially in the `doc_qa_task` where it involves using the `MemGPT` class, I noticed the import statement:
```python
from memgpt import MemGPT, utils
```
and the instantiation in the code:
```python
memgpt_client = MemGPT()
```
However, after a thorough search through the repository, I am unable to find the definition or implementation of the `MemGPT` class. This absence has made it challenging to understand how to proceed with setting up and running the experiments as intended.

Could you please provide guidance on where I might find the `MemGPT` class, or if there are any additional steps or documentation I should refer to? It's possible that I might have overlooked something or there are updates that are not yet pushed to the repository.

Thank you for your assistance and for sharing your work, which has been incredibly insightful.

**Comment by MrFCow** [comment]
I believe the main export is changed, now the export is exposing the function "create_client" instead of the MemGPT class.  So you can create a client like:

```
from memgpt import create_client
chat_client = create_client()
agent_info = chat_client.create_agent(
			name="Test Agent",
			persona="You are an agent, your alias is Johnny \
			"
		)
msg="what's your name"
response = chat_client.user_message(agent_id=agent_id, message=msg)
```

**Comment by raimondasl** [comment]
There is a deeper issue here though:
If we call create_client() without URL, it creates LocalClient. However LocalClient does not support llm_config or embedding_config in create_agent() method that are being passed in lines 97-106 of nested_kv.py and lines 184-191 of doc_qa.py:

`# Create a new Agent that models the scenario setup
    agent_state = memgpt_client.create_agent(
        {
            "name": agent_name,
            "persona": NESTED_PERSONA,
            "human": NESTED_HUMAN,
            "llm_config": config.default_llm_config,
            "embedding_config": config.default_embedding_config,
        }
    )`
    
OTOH if we pass URL to create_client(), it would create RESTClient. RESTClient supports llm_config or embedding_config but does not allow to override them and fails in the same place.

So, in short, current implementation does not support running nested_kv.py or doc_qa.py at all. :(
Can this be fixed?
I'd be happy to try fixing it if I got pointed in the right direction. It's currently not obvious to me if the right solution is some kind of variation of LocalClient that supports these config parameters or if we can allow RESTClient to override the configs.

Thanks.

BTW, there's a bunch of other issues with these scripts. I have fixed some of them. If there is interest, I could fork and submit PR. However without a fix to this issue my other fixes are not really tested.

**Comment by OswaldoBornemann** [comment]
The same questions that I met. How could we run the `paper_experiments`? @sarahwooders @cpacker

**Comment by cpacker** [comment]
@OswaldoBornemann can you try using the commit where the experiments were added to the repo [f239041](https://github.com/cpacker/MemGPT/commit/f2390418c4d4902a6ac98f7560fb774ea66855e6)?

The MemGPT codebase is under active development and unfortunately we don't have the bandwidth to keep the experiments code for the original research paper maintained, so rolling back the state of the repo to the relevant commit is going to be your best bet (at some point we may consider moving the experiments code out of the main repo into a frozen fork to prevent confusion). We definitely welcome any PRs to update the experiment scripts to the latest API though!

**Comment by OswaldoBornemann** [comment]
@cpacker @sahusiddharth  Thanks a lot. How to evaluate the accuracy of the nested_kv_task ?
